core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
+    if (removeRequestAcknowledged) {
+      val numExistingExecutors = allocationManager.executorIds.size - executorsPendingToRemove.size
+      var index = 0
+      for(index <- 0 until executorIdsToBeRemoved.size) {
+        logInfo(s"Removing executor " + executorIdsToBeRemoved(index) + " because it has been " +
```

Comment:
This mixes string interpolation and concatenation. You can rewrite it with interpolation, possibly defining extra vals for clarity.